pytorch_pfn_extras/training/extensions/log_report.py

```diff
@@ -253,3 +253,7 @@ def to_dataframe(self):
                 "Need to install pandas to use `to_dataframe` method."
             )
         return pandas.DataFrame(self._log_looker.get())
+
+    def finalize(self) -> None:
+        if self._writer is not None and hasattr(self._writer, 'finalize'):
```

Comment:
As `finalize` is the interface defined in the base writer class, let's call it directly rather than checking with hasattr.

Reply:
good catch, this was my mistake as I just copied the function straight from the snapshot extension. Fixed now! 🙇‍♂️